src/cmap/auth/gssapi.ts

```diff
@@ -15,7 +20,10 @@ import * as dns from 'dns';
 export class GSSAPI extends AuthProvider {
   auth(authContext: AuthContext, callback: Callback): void {
     const { connection, credentials } = authContext;
-    if (credentials == null) return callback(new MongoDriverError('credentials required'));
+    if (credentials == null)
+      return callback(
+        new MongoMissingCredentialsError('credentials required for gssapi authentication')
```

Comment:
Does this also apply to errors that start by referencing a variable name? If so, does that mean that we should restructure the message so that this is not the case?

For example, in `src/operations/update.ts: 131` we have the line:

```javascript
callback(new MongoCompatibilityError('arrayFilters are only supported in MongoDB 3.6+'));
```

Should this be changed to something like: 

```javascript
callback(new MongoCompatibilityError('The "arrayFilters" option is only supported in MongoDB 3.6+'));
```

Reply:
I guess that would be an exception to the rule, fine to leave that lowercase